File: Atm.java
package atm;

import java.util.Scanner;

public class Atm {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Account account = new Account(1000); // Initial balance

        System.out.println("Welcome to the ATM!");
        while (true) {
            System.out.println("Choose an option: 1. Check Balance 2. Deposit 3. Withdraw 4. Exit");
            int choice = scanner.nextInt();
            switch (choice) {
                case 1:
                    System.out.println("Your balance is: $" + account.getBalance());
                    break;
                case 2:
                    System.out.print("Enter amount to deposit: ");
                    double depositAmount = scanner.nextDouble();
                    account.deposit(depositAmount);
                    System.out.println("Deposited successfully! New balance: $" + account.getBalance());
                    break;
                case 3:
                    System.out.print("Enter amount to withdraw: ");
                    double withdrawAmount = scanner.nextDouble();
                    if (account.withdraw(withdrawAmount)) {
                        System.out.println("Withdrawal successful! New balance: $" + account.getBalance());
                    } else {
                        System.out.println("Insufficient balance!");
                    }
                    break;
                case 4:
                    System.out.println("Thank you for using the ATM. Goodbye!");
                    scanner.close();
                    return;
                default:
                    System.out.println("Invalid option. Please try again.");
            }
        }
    }
}

File: ATM.js
let balance = 1000.00;

function updateBalance() {
    document.getElementById('balance-amount').textContent = balance.toFixed(2);
}

function deposit() {
    document.getElementById('form-container').style.display = 'block';
    document.getElementById('amount').value = '';
    document.getElementById('amount').setAttribute('placeholder', 'Enter deposit amount');
}

function withdraw() {
    document.getElementById('form-container').style.display = 'block';
    document.getElementById('amount').value = '';
    document.getElementById('amount').setAttribute('placeholder', 'Enter withdrawal amount');
}

function submitTransaction() {
    const amount = parseFloat(document.getElementById('amount').value);
    if (isNaN(amount) || amount <= 0) {
        alert('Please enter a valid amount.');
        return;
    }

    const action = document.getElementById('amount').getAttribute('placeholder').includes('deposit') ? 'deposit' : 'withdraw';
    
    if (action === 'deposit') {
        balance += amount;
        alert(`You deposited $${amount.toFixed(2)}.`);
    } else {
        if (balance >= amount) {
            balance -= amount;
            alert(`You withdrew $${amount.toFixed(2)}.`);
        } else {
            alert('Insufficient funds.');
        }
    }

    updateBalance();
    cancelTransaction();
}

function cancelTransaction() {
    document.getElementById('form-container').style.display = 'none';
}
